### PR TITLE
Add uncertainty training

### PR DESCRIPTION
**Pull request by FeGeyer**
- Add the architecture for uncertainty training
- Added Olivia's contributions
- Removed unused keywords

### CODE REVIEW

File: radionets/dl_framework/callbacks.py [hunk 1]
```diff
@@ -249,3 +260,86 @@ def after_epoch(self):
...
+        self.arch_name = arch_name
+        self.amp_phase = amp_phase
+
+    def before_backward(self):
+        raise CancelBackwardException
+
+    def after_cancel_backward(self):
+        self.learn.loss.backward()
+
+        # access gradients of weights of layers (with specified batch and epoch)
```
Comment (ArnePoggenpohl) [inline radionets/dl_framework/callbacks.py:293]: Is all of that uncommented on purpose?

File: radionets/dl_framework/model.py [hunk 2]
```diff
@@ -401,3 +397,20 @@ def _conv_block(self, ni, nf, stride):
...
+    upper_half = x[:, :, 0 : x.shape[2] // 2, :].clone()
+    upper_left = upper_half[:, :, :, 0 : upper_half.shape[3] // 2].clone()
+    upper_right = upper_half[:, :, :, upper_half.shape[3] // 2 :].clone()
+    a = torch.flip(upper_left, dims=[-2, -1])
+    b = torch.flip(upper_right, dims=[-2, -1])
+
+    upper_half[:, :, :, 0 : upper_half.shape[3] // 2] = b
+    upper_half[:, :, :, upper_half.shape[3] // 2 :] = a
+
+    # a = torch.zeros((x.shape))
```
Comment (ArnePoggenpohl) [inline radionets/dl_framework/model.py:412]: Same as above. Can uncommented rows be removed?

File: radionets/dl_framework/callbacks.py (hunk 1 above)
Comment (FeGeyer) [reply radionets/dl_framework/callbacks.py:293]: Uncommented in last commit

File: radionets/dl_framework/model.py (hunk 2 above)
Comment (FeGeyer) [reply radionets/dl_framework/model.py]: Deleted

File: radionets/evaluation/scripts/start_evaluation.py [hunk 3]
```diff
@@ -102,6 +102,29 @@ def main(configuration_path):
...
+            arch_name=eval_conf["arch_name"],
+        )
+        output = output.save_output_pred()
+        grads_x, grads_y = output
+
+        # specify names of saved gradients in x and y
+        np.savetxt("grads_x.csv", grads_x, delimiter=",")
+        np.savetxt("grads_y.csv", grads_y, delimiter=",")
+
+        # # save image (no gradients)
```
[the remark this answers is not in the document]
Comment (FeGeyer) [reply radionets/evaluation/scripts/start_evaluation.py:119]: Also uncommented

**Review by Kevin2 — DISMISSED** [review]

**Review by Kevin2 — DISMISSED** [review]

**Review by Kevin2 — APPROVED** [review]

**Pull request merged**